index.js

```diff
@@ -44,12 +54,17 @@ class KintoServer {
     if (this.process) {
       throw new Error("Server is already started.");
     }
-    // Add the provided environment variables to the child process environment.
+    this.logs = [];
     // Keeping parent's environment is needed so that pserve's executable
     // can be found (with PATH) if KINTO_PSERVE_EXECUTABLE env variable was
     // not provided.
-    this.logs = [];
-    env = Object.assign({}, process.env, env);
+    // However, Kinto's config parsing logic tries to interpolate any
+    // %-code in any environment variable, so rather than polluting
+    // the environment with everything, just copy the variables we
+    // think will be necessary.
+    const sanitizedEnv = copyExisting(process.env, ["PATH", "VIRTUAL_ENV"]);
+    // Add the provided environment variables to the child process environment.
+    env = Object.assign({}, sanitizedEnv, env);
```

Comment:
I don't want to pass any `undefined`\ s, especially if we add any other variables that might or might not be present. `VIRTUAL_ENV` could be undefined if you're running outside a venv (although I admit that I generally don't).